org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/configuration/BuildConfiguration.java

```diff
@@ -33,5 +33,7 @@
 
     boolean isOfflineMode();
 
+    boolean isAutoRefresh();
```

Comment:
There are multiple names for this throughout the PR. I think we should call the property `autoSync` everywhere and the public method name should be called `isAutoSync()`. Please update the names like `autoRefresh` or `autoSyncEnabled`.